tuna/example/example_lib.py

```diff
@@ -98,65 +105,64 @@ def launch_worker(self, gpu_idx, f_vals, worker_lst):
     worker_lst.append(worker)
     return True
 
-  def compose_worker_list(self, machines):
+  def compose_worker_list(self, machines) -> Optional[List[ExampleWorker]]:
     # pylint: disable=too-many-branches
     """! Helper function to compose worker_list
-      @param res DB query return item containg available machines
-      @param args The command line arguments
+      @param machines containg available machines
+      @returns list of actual available machine list
     """
-    worker_lst = []
+    worker_lst: List[ExampleWorker] = []
     for machine in machines:
       if self.args.restart_machine:
         machine.restart_server(wait=False)
         continue
 
       #determine number of processes by compute capacity
       # pylint: disable=duplicate-code
-      worker_ids = super().get_num_procs(machine)
-      if len(worker_ids) == 0:
+      worker_ids: Union[Iterable, None] = super().get_num_procs(machine)
+      if (isinstance(worker_ids, Sequence) and len(worker_ids) == 0 and
```

Comment:
I think 
```
if not worker_ids:
  return None
```
would suffice, but I'd let @cderb comment on this

Reply:
@technicalgrp89  Come to think of it, your change to the if condition might break the code.

Reply:
When we change worker_id: list to Union[Iterable, None] , leads to error "it is not sized" parameter. to resolve the error, I had to introduce the check of isinstance(worker_ids, Sequence) and len(worker_ids) == 0.

Reply:
> @technicalgrp89 Come to think of it, your change to the if condition might break the code.

CI is passing.

Reply:
> When we change worker_id: list to Union[Iterable, None] , leads to error "it is not sized" parameter. to resolve the error, I had to introduce the check of isinstance(worker_ids, Sequence) and len(worker_ids) == 0.
> 
> @msaudulhassan how about keeping the worker_id:list. this would not raise any mypy error and CI is also passing

![image](https://github.com/ROCmSoftwarePlatform/MITuna/assets/68208941/9f99a6d3-dc3f-4859-bc32-80d1a267342a)

> > @technicalgrp89 Come to think of it, your change to the if condition might break the code.
> 
> CI is passing.

Double check with Chris

Reply:
Here's a compromise:
got to the implementation of get_num_procs
cast the return value to a list
Instead of returning None, return an empty list []